internal/policies/fetcher.go

```diff
@@ -38,27 +35,48 @@ func NewFetcher() (*Fetcher, error) {
 	return &Fetcher{client: client, filter: filterAuditablePolicies}, nil
 }
 
-// TODO implement this for all ns
-func (f *Fetcher) GetPoliciesForAllNamespaces() ([]policiesv1.Policy, error) {
-	return nil, errors.New("scanning all namespaces is not implemented yet. Please pass the --namespace flag to scan a namespace")
+// GetPoliciesForAllNamespace gets all auditable policies, and the number of
+// skipped policies
+// TODO implement this in the future
+func (f *Fetcher) GetPoliciesForAllNamespaces() ([]policiesv1.Policy, int, error) {
+	return nil, 0, errors.New("scanning all namespaces is not implemented yet. Please pass the --namespace flag to scan a namespace")
 }
 
-// GetPoliciesForANamespace gets all auditable policies for a given namespace
-func (f *Fetcher) GetPoliciesForANamespace(namespace string) ([]policiesv1.Policy, error) {
+// GetPoliciesForANamespace gets all auditable policies for a given namespace, and the number
+// of skipped policies
+func (f *Fetcher) GetPoliciesForANamespace(namespace string) ([]policiesv1.Policy, int, error) {
 	namespacePolicies, err := f.findNamespacesForAllClusterAdmissionPolicies()
 	if err != nil {
-		return nil, fmt.Errorf("can't fetch ClusterAdmissionPolicies: %w", err)
+		return nil, 0, fmt.Errorf("can't fetch ClusterAdmissionPolicies: %w", err)
 	}
 	admissionPolicies, err := f.getAdmissionPolicies(namespace)
 	if err != nil {
-		return nil, fmt.Errorf("can't fetch AdmissionPolicies: %w", err)
+		return nil, 0, fmt.Errorf("can't fetch AdmissionPolicies: %w", err)
 	}
 	for _, policy := range admissionPolicies {
 		policy := policy
 		namespacePolicies[namespace] = append(namespacePolicies[namespace], &policy)
 	}
 
-	return f.filter(namespacePolicies[namespace]), nil
+	filteredPolicies := f.filter(namespacePolicies[namespace])
+	skippedNum := len(namespacePolicies[namespace]) - len(filteredPolicies)
+	return filteredPolicies, skippedNum, nil
+}
+
+func (f *Fetcher) GetNamespace(nsName string) (*v1.Namespace, error) {
+	namespace := &v1.Namespace{}
+	err := f.client.Get(context.Background(),
+		client.ObjectKey{
+			Name: nsName,
+		},
+		namespace)
+	if err != nil && errorsApi.IsNotFound(err) {
+		return nil, fmt.Errorf("namespace not found: %s", nsName)
+	}
+	if err != nil {
+		return nil, fmt.Errorf("can't get namespace: %s", nsName)
```

Comment:
We should not hide the error, otherwise it will be hard to debug stuff

Reply:
Done in https://github.com/kubewarden/audit-scanner/pull/40/commits/2db512b545e1262c1288a96bc55252ee2ac676ca.